### PR TITLE
ci: Fail QNS run on Rust panic

### PR DESCRIPTION
**Pull request by larseggert**
So we hopefully catch interop panics in CI earlier.

### CODE REVIEW

**Review by KershawChang — APPROVED** [review]

**Comment by larseggert** [comment]
This needs to wait until #1975 is fixed, otherwise all CI will fail.

File: .github/actions/quic-interop-runner/action.yml [hunk 1]
```diff
@@ -72,6 +72,15 @@ runs:
         # Don't fail CI if the interop test fails
```
Comment (mxinden) [inline .github/actions/quic-interop-runner/action.yml:72]: Do I understand correctly that long-term we would want to fail CI?

**Review by mxinden — APPROVED** [review]
Thanks :+1:

File: .github/actions/quic-interop-runner/action.yml (hunk 1 above)
Comment (larseggert) [reply .github/actions/quic-interop-runner/action.yml:72]: Yes, once we get to the point that we consistently pass all tests (and all other stacks do to, when interop'ing with us).

Comment (mxinden) [reply .github/actions/quic-interop-runner/action.yml:72]: > (and all other stacks do to, when interop'ing with us).

We might want to pin their Docker images, such that a bug on their end doesn't break our CI. Not critical, thus I suggest keeping as is until this happens for the first time.

**Comment by larseggert** [comment]
The single remaining failure is [Run QNS (neqo-latest vs. neqo)](https://github.com/mozilla/neqo/actions/runs/9979399509/job/27578579516#logs), which is because mainline neqo doesn't have #1981 yet.

**Pull request merged**